features/253-auth-sdk-callback-in-next/api.md

```diff
@@ -0,0 +1,393 @@
+
+```json=
+{
+    "_access": null,
+    "_created_at": "<datetime>",
+    "_created_by": "<user_id>",
+    "_id": "user\/<record_id>",
+    "_ownerID": "<user_id>",
+    "_recordID": "<record_id>",
+    "_recordType": "user",
+    "_type": "record",
+    "_updated_at": "<datetime>",
+    "_updated_by": "<user_id>",
+    "email": "abc@example.com"
+}
+```
+
+To handle the case of record gear missing, auth gear will add a new table to save user's profile.
+
+```sql=
+CREATE TABLE _auth_user_profile (
+  user_id text REFERENCES _core_user(id),
+  created_at timestamp without time zone NOT NULL,
+  created_by text,
+  updated_at timestamp without time zone NOT NULL,
+  updated_by text,
+  data jsonb,
+  PRIMARY KEY(user_id),
+  UNIQUE (user_id)
+);
+```
+
+Auth gear will also have a store `SimpleUserProfileStore` which confirms `UserProfileStore` interface:
+
+Old:
+```go=
+type UserProfileStore interface {
+  CreateUserProfile(userProfile interface{}) error
+  GetUserProfile(userID string, userProfile *interface{}) error
+}
+```
+
+New:
+```go=
+// Meta is meta data part of a user profile record
+type Meta struct {
+  ID         string                 `json:"_id"`
+  Type       string                 `json:"_type"`
+  RecordID   string                 `json:"_recordID"`
+  RecordType string                 `json:"_recordType"`
+  Access     map[string]interface{} `json:"_access"`
+  OwnerID    string                 `json:"_ownerID"`
+  CreatedAt  time.Time              `json:"_createdAt"`
+  CreatedBy  string                 `json:"_createdBy"`
+  UpdatedAt  time.Time              `json:"_updatedAt"`
+  UpdatedBy  string                 `json:"_updatedBy"`
+}
+
+// Data refers the profile info of a user,
+// like username, email, age, phone number
+type Data map[string]interface{}
+
+// UserProfile refers user profile data type
+type UserProfile struct {
+  Meta
+  Data
+}
+
+type Store interface {
+  CreateUserProfile(userID string, data Data) (UserProfile, error)
+  GetUserProfile(userID string) (UserProfile, error)
+}
+```
+
+And, use `NewAuthResponse` to generate auth response,
+
+Old:
+
+```go=
+func NewAuthResponse(authInfo authinfo.AuthInfo, user skydb.Record, accessToken string) AuthResponse
+```
+
+New:
+
+```go=
+func NewAuthResponse(authInfo authinfo.AuthInfo, userProfile userprofile.UserProfile, accessToken string) AuthResponse
+```
+
+# More usage example
+
+## login
+
+```javascript=
+skygear.auth.signupWithUsername(username, password).then((user) => {
+  console.log(user); // user is an user record
+  console.log(user["username"]); // username of the user
+}, (error) => {
+  ;
+});
+```
+
+## update user profile
+
+```javascript=
+var user = skygear.auth.currentUser;
+user["username"] = "new-username";
+skygear.auth.updateUser(user).then((user) => {
+  console.log(user); // updated user record
+  console.log('Username is changed to: ', user["username"]);
+  return skygear.auth.whoami();
```

Comment:
is `skygear.auth.whoami()` needed? Or `updateUser` shall update the current user automatically?